src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs

```diff
@@ -384,6 +384,8 @@ public sealed class SourcesCommandConfiguration
         public string Username { get; set; }
         public string Password { get; set; }
         public int Priority { get; set; }
+        public string Certificate { get; set; }
+        public string CertificatePassword { get; set; }

```

Comment:
No, like with proxy authentication, you can have different credentials for for different nodes along the way (btw, the certificate password is not and should not be sent along - it's just to open local certificate file). My use case - reverse proxy configured to do TLS and authenticate with certificates - server & client certificates exchange establishes HTTPS, NuGet server behind it - authorizes. Given that HTTPS termination is commonly done with the proxy, the HTTP server behind it won't even get a say in the negotiations.
Hope that makes sense.